packages/cypress/src/utils/config.ts

```diff
@@ -29,27 +32,48 @@ export async function addDefaultE2EConfig(
   if (testingTypeConfig.length === 0) {
     const configValue =
       options.bundler === 'vite'
-        ? `nxE2EPreset(__filename, { cypressDir: '${options.directory}', bundler: 'vite' })`
-        : `nxE2EPreset(__filename, { cypressDir: '${options.directory}' })`;
+        ? `nxE2EPreset(__filename, { cypressDir: '${options.cypressDir}', bundler: 'vite' })`
+        : `nxE2EPreset(__filename, { cypressDir: '${options.cypressDir}' })`;
 
     updatedConfigContents = tsquery.replace(
       cyConfigContents,
       `${TS_QUERY_EXPORT_CONFIG_PREFIX} ObjectLiteralExpression:first-child`,
       (node: ObjectLiteralExpression) => {
+        let baseUrlContents = baseUrl ? `,\nbaseUrl: '${baseUrl}'` : '';
```

Comment:
Is there a case where `baseUrl` is falsy? Seems like it should always be defined?

Reply:
It is undefined when you provide a devServerTarget. In those cases, we grab the baseUrl from the devServerTarget